processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -1026,10 +1051,21 @@ public boolean matches()
       }
     }
 
-    // Currently unused, except by unit tests, since filters always support bitmap indexes currently.
-    // This will change when non-String dimensions are added.
     @Override
-    public ValueMatcher makeValueMatcher(String dimension, final Predicate predicate)
+    public ValueMatcher makeValueMatcher(String dimension, final DruidPredicateFactory predicateFactory)
+    {
+      ValueType type = getTypeForDimension(dimension);
+      switch (type) {
+        case LONG:
+          return makeLongValueMatcher(dimension, predicateFactory.makeLongPredicate());
+        case STRING:
+          return makeStringValueMatcher(dimension, predicateFactory.makeStringPredicate());
+        default:
+          throw new UOE("Cannot make ValueMatcher for type[%s]", type);
```

Comment:
@gianm I've changed these areas and the  ColumnSelectorBitmapIndexSelector to treat columns with unfilterable types (e.g., doubles, complex) as null-columns.

A test InvalidFilteringTest has been added to check this behavior